c7n/resources/iam.py

```diff
@@ -774,6 +774,120 @@ def _user_mfa_devices(resource):
         return matched
 
 
+@User.filter_registry.register('username')
+class UserName(ValueFilter):
```

Comment:
@kapilt thanks for the feedback! I'll try to see if I can filter the same way using a JMESPath in the key field.